app/controllers/products_controller.rb

```diff
@@ -26,28 +26,32 @@ def show
 
   def create
     @barcode = params[:barcode]
-
     @barcode = convert_to_jancode unless jancode_format?
 
     if in_database?
       @product = Product.find_by(barcode: @barcode)
-      redirect_to product_path(@product), notice: 'Barcode was scanned successfully.'
-    else
-      jancode_scraper = ScrapeJancodeService.new(@barcode)
-      jancode_scraper.call
-      @product = jancode_scraper.create_product
-
-      if @product.save
-        jancode_scraper.upload_image
-        redirect_to product_path(@product), notice: 'Barcode was scanned successfully.'
+      if comparing_scanned?
+        compare_scanned
       else
-        redirect_to root_path, alert: "Failed to scan barcode. #{@product.errors.full_messages.join(', ')}."
+        redirect_to product_path(@product)
       end
+    else
+      create_scraped_product
     end
   end
 
   private
 
+  def comparing_scanned?
+    params[:compare]
```

Comment:
Currently, both 'scan' and 'compare' buttons in the footer will lead you to the product comparison page. Because HTTP requests from the frontend will store value in params as string, thus `params[:compare] = "false"` is truthy and will make the code at line 74 go to the `compare_scan` no matter what value is inside as long as it has the key.

Reply:
```suggestion
    params.key?(:compare) && params[:compare] == 'true'
```

Reply:
Good catch!